.github/workflows/e2e.yaml

```diff
@@ -26,8 +26,16 @@ jobs:
       matrix:
         # TODO: Document which versions we support and cover them.
         # https://github.com/StackStorm/stackstorm-k8s/issues/342
+        # https://github.com/k3s-io/k3s/releases
         k3s-channel:
+          # - "v1.18.13+k3s1"
+          # - "v1.19.5+k3s1"
+          # - "v1.20.2+k3s1"
+          # - "v1.21.0+k3s1"
+          # - "v1.23.6+k3s1"
+          - "v1.24.4+k3s1"
           - "v1.25.4+k3s1"
+          - "v1.26.0+k3s1"
```

Comment:
I think having a K8s compatibility matrix would be more suitable for something like minikube as that better mimics the real K8s cluster and k3s is more customized and we may want just to check if the latest k3s works.

Reply:
I don't quite follow the argument here ... we're not proving whether or not k3s works ... and I don't know what you mean by "better mimics a real K8s cluster". According to [this](https://docs.k3s.io/) "K3s is a fully compliant Kubernetes distribution".

Reply:
The point of this matrix is to verify that `stackstorm-k8s` works under Kubernetes 1.24, 1.25, and 1.26.

Reply:
k3s is a modified lightweight 3rd-party distribution of K8s.
While it should be fully compatible, the reality there are always unique issues someone can hit running K3s compared to running an official K8s cluster. See: https://github.com/k3s-io/k3s/issues

`minikube` is re-using `kubeadm` and other primitives under the hood and is more comparable to running an official k8s cluster.

StackStorm is not an IoT platform and ideally if we test against environments closer to its real-world usage. k3s is ok for experimental compatibility check, but I'd prefer if we run the K8s build matrix with `minikube`.

Here is the ideal build matrix: 
- minikube-k8s-v1.24
- minikube-k8s-v1.25
- minikube-k8s-latest
- k3s-experimental-latest

Hope that gives more context ;)

Reply:
We'll just have to agree to disagree, `k3s` is [certified to be conformant](https://www.cncf.io/certification/software-conformance/).

Reply:
There is always a diff and unique issues that are present in K3s, but not in the K8s and vice-versa.

If something fails in tests in K3s that does work with `minikube` or prod K8s cluster I'd like to avoid the rabbit hole and support burden for us. And vice-versa, if tests passed with the k3s distro and the user comes up with the issue that something didn't work in the official K8s, - I don't want it for maintainers debugging this.

I'd like to be closer in tests to the real K8s environment where StackStorm is most likely to be deployed, and we can do it already with minikube that we're already running.

Reply:
BTW we tried KinD (another conformant distro), had some issues in the past, but it might work better now.

Do you have specific requirements to run and test StackStorm under the `k3s` distro, something that's required by your environment?

Or would you just like having different supported K8s versions in the build matrix?

Reply:
Personally, I do run `k3s` in production (and, I know others that do as well). I understand where you are coming from, and I'm fine with dropping `k3s` if that's what you want, but I'd kindly ask that you not slight their project by making statements that it's "not ready for production" or potentially only suitable for "edge or IoT". If you could spare the time, I think [this](https://traefik.io/glossary/k3s-explained/) is worthy of a quick read.

Reply:
When you said "kubeadm and other primitives under the hood and is more comparable to running an official k8s cluster" what I was thinking was "we're not testing the mechanism by which Kubernetes is stood up; what we should be testing is that stackstorm-ha successfully deploys to Kubernetes" -- and of course, the current test is doing that. In my mind, since the CNCF has officially adopted `k3s` as a distribution, I'll take that at face-value that it's (`k3s`) is a viable and *supported* distribution. Again, I personally don't care if we test against `k3s` or `minikube`, or both.

Reply:
When you mentioned to "See: https://github.com/k3s-io/k3s/issues", what'd I'd ask is, which of those issues specifically pertains to "incompatibility with upstream proper"?